Tests/Algorithm/Framework/Portfolio/AccumulativeInsightPortfolioConstructionModelTests.cs

```diff
@@ -0,0 +1,329 @@
+        {
+            SetPortfolioConstruction(language, _algorithm);
+
+            // First emit long term insight
+            var insights = new[] { GetInsight(Symbols.SPY, direction, _algorithm.UtcTime) };
+            var targets = _algorithm.PortfolioConstruction.CreateTargets(_algorithm, insights).ToList();
+            var expectedTargets = new List<IPortfolioTarget> { PortfolioTarget.Percent(_algorithm, Symbols.SPY, (int)direction * 1m * (decimal)DefaultPercent) };
+            AssertTargets(expectedTargets, targets);
+
+            // One minute later, emits insight to add to portfolio
+            SetUtcTime(_algorithm.UtcTime.AddMinutes(1));
+            insights = new[] { GetInsight(Symbols.SPY, direction, _algorithm.UtcTime) };
+            targets = _algorithm.PortfolioConstruction.CreateTargets(_algorithm, insights).ToList();
+
+            expectedTargets = new List<IPortfolioTarget> { PortfolioTarget.Percent(_algorithm, Symbols.SPY, (int)direction * 1m * 2 * (decimal)DefaultPercent) };
+            AssertTargets(expectedTargets, targets);
+
+            // One minute later, emits flat insight
+            SetUtcTime(_algorithm.UtcTime.AddMinutes(1));
+            insights = new[] { GetInsight(Symbols.SPY, InsightDirection.Flat, _algorithm.UtcTime) };
+            targets = _algorithm.PortfolioConstruction.CreateTargets(_algorithm, insights).ToList();
+
+            expectedTargets = new List<IPortfolioTarget> { PortfolioTarget.Percent(_algorithm, Symbols.SPY, (int)direction * 1m * (decimal)DefaultPercent) };
+
+            AssertTargets(expectedTargets, targets);
+        }
+
+
+        [Test]
+        [TestCase(Language.Python)]
+        [TestCase(Language.CSharp)]
+        public void WeightsProportionally(Language language)
+        {
+            SetPortfolioConstruction(language, _algorithm);
+
+            var insights = new[]
+            {
+                GetInsight(Symbols.SPY, InsightDirection.Down, _algorithm.UtcTime),
+                GetInsight(Symbol.Create("IBM", SecurityType.Equity, Market.USA),
+                    InsightDirection.Down, _algorithm.UtcTime)
+            };
+
+            // they will each share, proportionally, the total portfolio value
+            var amount = _algorithm.Portfolio.TotalPortfolioValue * (decimal)DefaultPercent;
+
+            var expectedTargets = _algorithm.Securities.Where(pair => insights.Any(insight => pair.Key == insight.Symbol))
+                .Select(x => new PortfolioTarget(x.Key, (int)InsightDirection.Down * Math.Floor(amount / x.Value.Price)));
+
+            var actualTargets = _algorithm.PortfolioConstruction.CreateTargets(_algorithm, insights).ToList();
+            Assert.AreEqual(2, actualTargets.Count);
+
+            AssertTargets(expectedTargets, actualTargets);
+        }
+
+        [Test]
+        [TestCase(Language.Python)]
+        [TestCase(Language.CSharp)]
+        public void GeneratesTargetsForInsightsWithNoConfidence(Language language)
```

Comment:
Some of these unit tests were designed for the weighted based portfolio construction models and won't really test this new one. It seems they should be replaced with a few specific unit tests, asserting on the accumulative behavior

Reply:
I went through and made sure they were all updated and applied to this portfolio construction model (and deleted the ones that don't). The confidence one should behave correctly here. It is twofold: a) it does test a new condition, b) I'm looking forward to inheriting from this parent model to create a smarter AccumulativeInsight model that can use confidence. In that model, a confidence of 0 should not generate a target. So we should see two behaviours uniquely tested where we ensure updating something in the future doesn't break the tests.